### DISCUSSION TITLE
API call to next.js backend before Router.push causes the route not to change

### DISCUSSION THREAD

**Discussion by JackOddy** [vercel/next.js#21070] [Help]
Hi all, hope someone can help me

I'm awaiting an API call to the next.js backend and then calling router.push to a new route. 

If the API call is commented out then the page changes correctly, however if it is there, it seems the `route-loader.js` which injects the script for the next page hangs forever. 
If i refresh the page, the router.push promise resolves and  do land on the new page.

It seems to specifically be an API call (we're usnig axios) to causes this effect, awating an empty promise before router.push didn't seem to have this effect. 

We have no idea what is causing this and thought it could be related to https://github.com/vercel/next-plugins/issues/282 some how, however this issue is closed (and we're using styled components).

Does anyone have any idea what might be causing this and what else we might look at?

also -- this seems to only be in dev mode, so perhaps related to compilation of the pages at run time?

**Comment by DmytroGolovin** [comment]
Hello, I am facing a simillar problem, did you find any solution?
